pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -425,10 +425,11 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
 
 		message, annotations := evictionMessage(resourceToReclaim, pod, statsFunc, thresholds, observations)
 		condition := &v1.PodCondition{
-			Type:    v1.DisruptionTarget,
-			Status:  v1.ConditionTrue,
-			Reason:  v1.PodReasonTerminationByKubelet,
-			Message: message,
+			Type:               v1.DisruptionTarget,
+			ObservedGeneration: podutil.GetPodObservedGenerationIfEnabledOnCondition(&pod.Status, pod.Generation, v1.DisruptionTarget),
```

Comment:
This is going to get a little messy, but I don't think we can use `pod.Status` here. This condition gets passed to `evictPod` which in turn calls `killPodFunc` with a status-updating method. This should use the status that's passed in to that function.

Here's what I think you should do:
1. Unconditionally set `ObservedGeneration: pod.Generation` here
2. Before `UpdatePodCondition` is called (line 622), update the observed generation with `GetPodObservedGenerationIfEnabledOnCondition`:
```diff
		if condition != nil {
+			condition.ObservedGeneration = podutil.GetPodObservedGenerationIfEnabledOnCondition(status, condition.ObservedGeneration, condition.Type)
			podutil.UpdatePodCondition(status, condition)
		}
```

Reply:
Alternatively, you could consider making this part of `UpdatePodCondition`, but that's a bigger change.